src/reader-activation/index.js

```diff
@@ -336,6 +336,21 @@ export function getAuthStrategy() {
 	return getCookie( 'np_auth_strategy' );
 }
 
+/**
+ * Debug logging function that only logs when localStorage flag is set.
+ *
+ * @param {string} level Log level ('log' or 'error').
+ * @param {...any} args  Arguments to pass to console.
+ */
+// eslint-disable-next-line no-console
+export function debugLog( level = 'log', ...args ) {
+	if ( localStorage.getItem( 'newspack-google-oauth-debug' ) === 'true' ) {
+		const method = level === 'error' ? 'error' : 'log';
+		// eslint-disable-next-line no-console
+		console[ method ]( ...args );
+	}
+}
```

Comment:
We can avoid code repetition by placing this in `src/reader-activation/utils.js` and importing.

We should also use a more generic name here; `newspack-google-oauth-debug` sounds too specific for this function.